client/src/includes/a11y-result.ts

```diff
@@ -70,6 +72,101 @@ export const getAxeConfiguration = (
   return null;
 };
 
+/**
+ * Custom rule for checking image alt text. This rule checks if the alt text for images
+ * contains file extensions or URLs, which are considered poor quality alt text.
+ * The rule will be added via the Axe.configure() API.
+ * https://github.com/dequelabs/axe-core/blob/master/doc/API.md#api-name-axeconfigure
+ */
+export const checkImageAltText = (node: Element) => {
+  const imageFileExtensions = [
+    'apng',
+    'avif',
+    'gif',
+    'jpg',
+    'jpeg',
+    'jfif',
+    'pjpeg',
+    'pjp',
+    'png',
+    'svg',
+    'tif',
+    'webp',
+  ];
+  const imageURLs = ['http://', 'https://', 'www.'];
+  const altTextAntipatterns = new RegExp(
+    `\\.(${imageFileExtensions.join('|')})|(${imageURLs.join('|')})`,
+    'i',
+  );
+
+  const image = node as HTMLImageElement;
```

Comment:
Checking if you tried just setting the `node` parameter type to `HTMLImageElement`?